Migrate to v0.5.x; Add chapter on Turing Completeness

felix91gr opened the pull request:
I was just adding the chapter on Turing Completeness (by reduction of Tag Systems into Declarative Macros), and whilst trying to build the book I realized that since the last time this book was deployed, a new minor version of mdBook had been released which broke the build.

Hence, this PR including 2 commits:
1. Remove `book.multilingual` and `book.edition` fields:
    * [`book.multilingual`](https://github.com/rust-lang/mdBook/blob/master/CHANGELOG.md#config-changes) has been removed.
    * `book.edition` seems to have been obsoleted by `rust.edition`. I wasn't able to find a note saying why the build would be breaking, but redundancy with the relatively new `rust.edition` field is my current guess as to why it was.
2. Add chapter on Turing Completeness.
    * The main contribution is the proof by reduction of Tag Systems to Declarative Macros.
    * Includes some context on Tag Systems and computational power
    * Ends with the consequences of being Turing Complete. Mostly from the perspective of tooling.

felix91gr committed:
Remove book.multilingual and book.edition fields
This is needed for migration to v0.5.x of mdBook

felix91gr committed:
Add chapter on turing completeness

felix91gr commented:
> _**Note:** it previously threw an error because I didn't specify which code snippets to ignore. That's fixed now._

@Veykril it's erroring on a code snippet that's part of the existing repo. Should I add a commit to fix that?

programmerjake commented on src/decl-macros/minutiae/turing-completeness.md:
```diff
@@ -0,0 +1,202 @@
+
+> **Note**: it's the same as noting that a computer's RAM is finite, and thus doesn't represent a real Turing Machine.
+
+Besides, this limit can be bypassed by setting the following attribute: `#![recursion_limit = "256"]`
```
256 is also a pretty low limit, maybe replace with `<N>` and explain that you can put an arbitrarily-large number there (within rustc's internal integer type limits, idk which type it uses there)

felix91gr replied:
Indeed, good catch! I've updated it to be more explicit about the attribute being something arbitrary one can choose.

(I tried a couple of large numbers, and it definitely seems like it uses something larger than `u32::MAX`! The limit seems to be pretty large)

Veykril commented:
Yea if you could just fix that as well that would be appreciated!

felix91gr committed:
Remove trailing semicolon for future compatibility
Done to comply with #[deny(semicolon_in_expressions_from_macros)],
which itself is a consequence of #[deny(future_incompatible)]

felix91gr commented:
@Veykril all done fam :)

Veykril reviewed (approved):
Thanks!
